app/LogModel.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace API;
+
+use DateTime;
+use Illuminate\Http\Request;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\DatabaseManager as DB;
+
+class LogModel extends Model
+{
+    protected $database;
+
+    //Instantiate new object for Result class
+    function __construct(DB $db)
+    {
+    	$this->database = $db;
+    }
+
+    /*
+     *This function is used to make an entry in database 
+     *to keep a record of all the search made by different user.
+     *@param string---src-----type of search made by user.
+     *@param string---query---actual query made by the user.
+     *@param int------userId--user id of logged in user.
+    */
+    public function logTableEntry($src, $query, $userId)
```

Comment:
What sort of comment is that. -- was an indication that those are extra space to be removed. Google, read and fix it.
eg -
    /**
     * Create a new event instance.
     *
     * @param  string  $connectionName
     * @param  \Illuminate\Contracts\Queue\Job  $job
     * @param  array  $data
     * @param  \Throwable  $exception
     * @return void
     */